src/modules/commitment/main_impl.h

```diff
@@ -263,4 +263,55 @@ int secp256k1_pedersen_blind_generator_blind_sum(const secp256k1_context* ctx, c
     return 1;
 }
 
+/* Generates a blinding key that contains a hashed switch commitment. */
+int secp256k1_blind_switch(const secp256k1_context* ctx, unsigned char* blind_switch, const unsigned char* blind, uint64_t value, const secp256k1_generator* value_gen, const secp256k1_generator* blind_gen, const secp256k1_pubkey* switch_pubkey) {
+    secp256k1_sha256 hasher;
+    secp256k1_pedersen_commitment commit;
+    unsigned char buf[33];
+    size_t buflen = sizeof(buf);
+    unsigned char hashed[32];
+    int overflow;
+    secp256k1_scalar blind_switch_scalar;
+    secp256k1_pubkey tmp_pubkey;
+    secp256k1_scalar tmp_scalar;
+    VERIFY_CHECK(ctx != NULL);
+    ARG_CHECK(blind_switch != NULL);
+    ARG_CHECK(blind != NULL);
+    ARG_CHECK(value_gen != NULL);
+    ARG_CHECK(blind_gen != NULL);
+    ARG_CHECK(switch_pubkey != NULL);
+
+    secp256k1_sha256_initialize(&hasher);
+    if (secp256k1_pedersen_commit(ctx, &commit, blind, value, value_gen, blind_gen) != 1) {
+        return 0;
+    }
+    if (secp256k1_pedersen_commitment_serialize(ctx, buf, &commit) != 1) {
+        return 0;
+    }
+    secp256k1_sha256_write(&hasher, buf, buflen);
+
+    memcpy(&tmp_pubkey, switch_pubkey, sizeof(*switch_pubkey));
```

Comment:
how about `tmp_pubkey = *switch_pubkey`?  maybe easier to understand.

Reply:
We shouldn't do this because `secp256k1_ec_pubkey_tweak_mul` modifies the input public key point, and we want to leave the input unchanged, so it can be used again on future calls. This means we have to copy it

Reply:
`mp_pubkey = *switch_pubkey` means a value copy, not a pointer assignment.
`secp256k1_ec_pubkey_tweak_mul` modify `tmp_pubkey`.

Reply:
Sorry, you are right. I was confusing myself :) Will change it